test/threadsafe_function/threadsafe_function_sum.cc

```diff
@@ -13,12 +13,21 @@ namespace {
 struct TestData {
 
   TestData(Promise::Deferred&& deferred) : deferred(std::move(deferred)) {};
+
+  // These variables are accessed only from the main thread. They keep track of
+  // the number of expected incoming calls that have completed. We do not want
+  // to release the thread-safe function until all expected calls are complete.
+  size_t threadsCreated = 0;
+  size_t callsCompleted = 0;
```

Comment:
This one doesn't seem to be only accessed from the main thread since they are accessed in `entryAcquire`?

Reply:
`callsCompleted` is not accessed in `entryAcquire` but in the lambda that ends up getting called on the main thread.

Reply:
Ah, that's right.